### PR TITLE
feat(policy): use pretrained vision encoder weights by default for diffusion and vqbet

### PR DESCRIPTION
**Pull request by villekuosmanen**
## Title

feat(policy): use pretrained vision encoder weights by default for diffusion and vqbet

## Type / Scope

- **Type**: Bug or Feature (depending on how you define it)
- **Scope**: Diffusion and VQBet default config changes

## Summary / Motivation

One common theme with LeRobot over the past few years has been the dominance of ACT as a stable baseline. Many people have found it difficult to train Diffusion policies to work for any realistic task besides PushT.
- https://github.com/huggingface/lerobot/issues/2796
- https://github.com/huggingface/lerobot/issues/2506
- https://github.com/huggingface/lerobot/issues/502
- https://github.com/huggingface/lerobot/issues/2761

After digging into this, I noticed the likely cause of all these failed training runs. The LeRobot diffusion policy (and VQBeT, for that matter) initialises with random weights for the image encoder by default!

I suspect this is an oversight in the config defaults, and while users can change these values by themselves I don't think many people have the expertise required to change the relevant fields, or even expect to have to change the pre-trained vision encoder params.

I have also changed `use_group_norm` to default to False for compatibility in the model init:
```
        if config.use_group_norm:
            if config.pretrained_backbone_weights:
                raise ValueError(
                    "You can't replace BatchNorm in a pretrained model without ruining the weights!"
                )
```

In addition for diffusion policy I propose increasing the default horizon and num_action_stps. The current defaults are likely too low for most real tasks and designed for the easier PushT task. I also propose setting the default for `use_separate_rgb_encoder_per_camera` to True, to better support setups with wrist and scene cameras. This is another field that most users will not know exists.

## Related issues

https://github.com/huggingface/lerobot/issues/2944

## What changed

- Short, concrete bullets of the modifications (files/behaviour).
- Short note if this introduces breaking changes and migration steps.

## How was this tested (or how to run locally)

Tested training a diffusion policy locally with pre-trained weights (and some other changes to benefit my dataset) and it works better and trains much faster than the default values though it's still far from perfect.

Example:

- Train diffusion for a real task with and without pre-trained image encoder weights and compare their behaviours.

## Checklist (required before merge)

- [ ] Linting/formatting run (`pre-commit run -a`)
- [ ] All tests pass locally (`pytest`)
- [ ] Documentation updated
- [ ] CI is green

## Reviewer notes

- Anything the reviewer should focus on (performance, edge-cases, specific files) or general notes.
- Anyone in the community is free to review the PR.

### CODE REVIEW

**Comment by paulpacaud** [comment]
I am also curious to understand the developer's choice to set the default config in ACT to `pretrained_backbone_weights: str | None = "ResNet18_Weights.IMAGENET1K_V1"` while to set the default config in DP is `pretrained_backbone_weights: str | None = None`.

However, it is not absurd to train the Diffusion Policy from scratch, as it is consistent with the original Diffusion Policy paper (ablation part) claims. In the paper:
- In simulation, the best config is = pretrained + finetuning ("we found finetuning the pretrained vision encoder with a small learning rate (10x smaller vs diffusion policy network) gives the best performance overall.")
- In the real world, the best config is end-to-end training from scratch ("We found that end-to-end training is still the most effective way to incorporate visual observation into Diffusion Policy, and our best-performing models were all end-to-end trained.")

So, using by default pretrained vision encoder weights in the diffusion policy config would kind of go against the original paper findings. But it would add consistency and a fairer comparison with ACT, which uses the pretrained weights.

**Comment by paulpacaud** [comment]
Just following up adding that I tested the horizon=64 and n_action_steps=32 on real-robot tasks with the SO100, and it indeed works better than the small 16/8 current default values that are extremely slow for real-robot evaluation.

**Comment by imstevenpmwork** [comment]
Given the results from: https://discord.com/channels/1216765309076115607/1216765309558722622/1500465687984996433
I'm okay with merging this PR as long as it has been tested. Will kick the CI in the meantime

Thanks for the contribution! 😄

**Review by imstevenpmwork — DISMISSED** [review]
LGTM, thanks!

**Review by imstevenpmwork — APPROVED** [review]
